lock-impl/src/main/java/com/palantir/lock/logger/LockServiceStateLogger.java

```diff
@@ -0,0 +1,216 @@
+
+        Path outputDirPath = Paths.get(outputDir);
+        Files.createDirectories(outputDirPath);
+
+        dumpYamlsInNewFiles(outputDir, generatedOutstandingRequests, generatedHeldLocks);
+    }
+
+    private Map<String, Object> generateOutstandingLocksYaml(Map<LockClient, Set<LockRequest>> outstandingLockRequestsMap) {
+        Map<String, SimpleLockRequestsWithSameDescriptor> outstandingRequestMap = Maps.newConcurrentMap();
+
+        outstandingLockRequestsMap.forEach((client, requestSet) -> {
+            if (requestSet != null) {
+                ImmutableSet<LockRequest> lockRequests = ImmutableSet.copyOf(requestSet);
+                lockRequests.forEach(lockRequest -> {
+                    List<SimpleLockRequest> requestList = getDescriptorSimpleRequestMap(client, lockRequest);
+                    requestList.forEach(request -> {
+                        outstandingRequestMap.putIfAbsent(request.getLockDescriptor(),
+                                new SimpleLockRequestsWithSameDescriptor(request.getLockDescriptor()));
+                        outstandingRequestMap.get(request.getLockDescriptor()).addLockRequest(request);
+                    });
+                });
+            }
+        });
+
+        List<SimpleLockRequestsWithSameDescriptor> sortedOutstandingRequests = sortOutstandingRequests(outstandingRequestMap.values());
+
+        Yaml yaml = new Yaml(getRepresenter(), getDumperOptions());
+        return nameObjectForYamlConvertion(sortedOutstandingRequests, OUTSTANDING_LOCK_REQUESTS_TITLE);
+    }
+
+    private List<SimpleLockRequestsWithSameDescriptor> sortOutstandingRequests(
+            Collection<SimpleLockRequestsWithSameDescriptor> outstandingRequestsByDescriptor) {
+
+        List<SimpleLockRequestsWithSameDescriptor> sortedEntries = Lists.newArrayList(outstandingRequestsByDescriptor);
+        sortedEntries.sort(
+                new Comparator<SimpleLockRequestsWithSameDescriptor>() {
+                    @Override
+                    public int compare(SimpleLockRequestsWithSameDescriptor o1,
+                            SimpleLockRequestsWithSameDescriptor o2) {
+                        return Integer.compare(o2.getLockRequestsCount(),
+                                o1.getLockRequestsCount());
+                    }
+                });
+        return sortedEntries;
+    }
+
+    private Map<String, Object> generateHeldLocks(ConcurrentMap<HeldLocksToken, LockServiceImpl.HeldLocks<HeldLocksToken>> heldLocksTokenMap) {
+
+        List<Map<String, Object>> mappedLockListGroupedByToken = heldLocksTokenMap.entrySet().stream()
+                .map(heldLocksTokenMapEntry ->
+                        getDescriptorToTokenMap(heldLocksTokenMapEntry.getKey(), heldLocksTokenMapEntry.getValue()))
+                .collect(Collectors.toList());
+
+        Map<String, Object> mappedLocksToToken = Maps.newConcurrentMap();
+        mappedLockListGroupedByToken.forEach(mappedLocksToToken::putAll);
```

Comment:
combine these lines into your mapping above so that it collapses into the map you want.  Again you don't need a concurrentmap here.